packages/dds/sequence/src/intervalIndex/overlappingSequenceIntervalsIndex.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	Client,
+	ISegment,
+	ReferenceType,
+	compareReferencePositions,
+} from "@fluidframework/merge-tree";
+import {
+	IIntervalHelpers,
+	IntervalType,
+	SequenceInterval,
+	createPositionReferenceFromSegoff,
+} from "../intervalCollection";
+import { IntervalTree } from "../intervalTree";
+import { SequenceIntervalIndexes } from "./SequenceIntervalIndexes";
+
+class OverlappingSequenceIntervalsIndex implements SequenceIntervalIndexes.Overlapping {
```

Comment:
You should be able to implement this by extending `OverlappingIntervalsIndex`. That will remove some of the remove/add boilerplate.

You'll need to export that from the file it's in. You might want to just move it now.